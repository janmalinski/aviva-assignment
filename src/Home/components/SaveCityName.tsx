import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

interface ISaveCityNameProps {
    cityNameSaved: boolean;
    disabled: boolean;
    handleSaveCity: () => void;
    testID?: string;
};

const SaveCityName = ({cityNameSaved, handleSaveCity, disabled, testID}: ISaveCityNameProps) => {
  return (
    <TouchableOpacity testID={testID} style={styles.row} onPress={handleSaveCity} disabled={disabled} >
        <View style={styles.checkboxContainer}>
        {cityNameSaved && <Ionicons testID='checkbox-icon' name={'checkmark-sharp'} size={12} />}
        </View>
        <Text style={styles.saveFavouriteCityNameText}>Save name of favourite city</Text>
  </TouchableOpacity>
  )
};

export default React.memo(SaveCityName);

const styles = StyleSheet.create({
    row: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    checkboxContainer: {
        height: 16,
        width: 16,
        borderWidth: 1,
        borderColor: '#000',
        justifyContent: 'center',
        alignItems: 'center',
        marginRight: 5
    },
    saveFavouriteCityNameText: {
        fontSize: 16
      },
})