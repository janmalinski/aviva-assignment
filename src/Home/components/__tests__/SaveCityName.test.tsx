import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import SaveCityName from '../SaveCityName';


const handleSaveCity = jest.fn();
const SaveCityNameEl = <SaveCityName testID='save-city-checkbox' disabled={false} cityNameSaved={false} handleSaveCity={handleSaveCity} />;

describe('Rendering SaveCityName', () => {
    it('Should match a snapshot', () => {
        const elJson = render(SaveCityNameEl).toJSON();
        expect(elJson).toMatchSnapshot(); 
    })
});

describe('Behaviour SaveCityName', () => {
    it('Should handle onPress event', () => {
        const { getByTestId} =  render(SaveCityNameEl);
        fireEvent.press(getByTestId('save-city-checkbox'));
        expect(handleSaveCity).toBeCalled();
    });

    it(`Should ignore onPress event if disabled`, () => {
        const mockFn = jest.fn();
        const { getByTestId} =  render(<SaveCityName testID='save-city-checkbox' disabled cityNameSaved={false} handleSaveCity={mockFn} />);
        fireEvent.press(getByTestId('save-city-checkbox'));
         expect(mockFn).not.toBeCalled();
      });

      it("Should display checkmark icon when cityNamedSaved is true", async() => {
        const { getByTestId } =  render(<SaveCityName testID='save-city-checkbox' disabled={false} cityNameSaved={true} handleSaveCity={handleSaveCity} />);
        fireEvent.press(getByTestId('save-city-checkbox'));
        await waitFor(() => getByTestId('checkbox-icon') )
      });

});