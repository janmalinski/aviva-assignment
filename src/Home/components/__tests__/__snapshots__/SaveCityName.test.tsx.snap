// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering SaveCityName Should match a snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="save-city-checkbox"
>
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "#000",
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "marginRight": 5,
        "width": 16,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Save name of favourite city
  </Text>
</View>
`;
